src/models/tiles/geometry/jxg-point.ts

```diff
@@ -1,68 +1,60 @@
-import { castArray } from "lodash";
-import { getColorMapEntry } from "../../shared/shared-data-set-colors";
+import { castArray, merge } from "lodash";
 import { uniqueId } from "../../../utilities/js-utils";
-import { JXGChangeAgent, JXGCoordPair, JXGUnsafeCoordPair } from "./jxg-changes";
+import { JXGChangeAgent, JXGCoordPair, JXGProperties, JXGUnsafeCoordPair } from "./jxg-changes";
 import { objectChangeAgent, isPositionGraphable, getGraphablePosition } from "./jxg-object";
 import { prepareToDeleteObjects } from "./jxg-polygon";
+import { fillPropsForColorScheme } from "./geometry-utils";
 
-// For snap to grid
-const kPrevSnapUnit = 0.2;
+// Set as snap unit for all points that have snapToGrid set.
+// Also used as the distance moved by arrow-key presses.
 export const kSnapUnit = 0.1;
 
-export const kPointDefaults = {
-              fillColor: "#0069ff",
-              strokeColor: "#000000",
-              selectedFillColor: "#ff0000",
-              selectedStrokeColor: "#ff0000"
-            };
+const defaultPointProperties = Object.freeze({
+  strokeColor: "#000000", highlightStrokeColor: "#0081ff",
+  strokeWidth: 1,         highlightStrokeWidth: 10,
+  strokeOpacity: 1,       highlightStrokeOpacity: .12,
+  fillOpacity: 1,         highlightFillOpacity: 1,
+  size: 4,
+  snapSizeX: kSnapUnit,
+  snapSizeY: kSnapUnit,
+  withLabel: true,
+  transitionDuration: 0
+});
 
-const defaultProps = {
-        fillColor: kPointDefaults.fillColor,
-        strokeColor: kPointDefaults.strokeColor
-      };
+const selectedPointProperties = Object.freeze({
+  strokeColor: "#0081ff", highlightStrokeColor: "#0081ff",
+  strokeWidth: 10,        highlightStrokeWidth: 10,
+  strokeOpacity: .25,     highlightStrokeOpacity: .25
+});
 
-// fillColor/strokeColor are ephemeral properties that change with selection;
-// we store the desired colors in clientFillColor/clientStrokeColor for persistence
-// colors for linked points are derived from the link color map
-export function syncClientColors(props: any) {
-  const { selectedFillColor, selectedStrokeColor, ...p } = props || {} as any;
-  const colorMapEntry = getColorMapEntry(p.linkedTableId);
+const phantomPointProperties = Object.freeze({
+  fillOpacity: .5,        highlightFillOpacity: .5,
+  withLabel: false
+});
```

Comment:
A number of these values, e.g. `"#0081ff"`, `10`, `.25`, are repeated multiple times in the code. Could reduce redundancy by defining some shared constants which would make it easier to change them down the road, but they're only used in a handful of well-defined places in the code, so I don't feel particularly strongly about it.